script.js

```diff
@@ -1,4 +1,493 @@
+    if (playerProfiles[i].playerRoundScore == "WIN") {
+      playerProfiles[i].playerChipsBalance +=
+        playerProfiles[i].playerCurrRoundBet;
+    }
+    if (playerProfiles[i].playerRoundScore == "LOSE") {
+      playerProfiles[i].playerChipsBalance -=
+        playerProfiles[i].playerCurrRoundBet;
+    }
+    if (playerProfiles[i].playerRoundScore == "BLACKJACK! - WIN") {
+      playerProfiles[i].playerChipsBalance +=
+        playerProfiles[i].playerCurrRoundBet * 2;
+    }
+  }
+
+  var playerHandmessage =
+    "Dealer cards: " +
+    dealerProfile.dealerHand +
+    "<br> Dealer current points: " +
+    dealerProfile.dealerHandValue +
+    "<br>";
+
+  for (i = 0; i < playerProfiles.length; i++) {
+    var playerHandmessage =
+      playerHandmessage +
+      "<br><br>Player " +
+      playerProfiles[i].id +
+      "'s cards:" +
+      playerProfiles[i].playerHand +
+      "<br> Current points: " +
+      playerProfiles[i].playerHandValue +
+      "<br> Chips balance: " +
+      playerProfiles[i].playerChipsBalance +
+      "<br> Bet this round: " +
+      playerProfiles[i].playerCurrRoundBet +
+      "<br> Result: " +
+      playerProfiles[i].playerRoundScore;
+  }
+  for (var i = 0; i < playerProfiles.length; i++) {
+    if (playerProfiles[i].playerChipsBalance < 1) {
+      playerProfiles.splice(i, 1);
+      numPlayers -= 1;
+    }
+  }
+  return playerHandmessage;
+}
+function resetGame() {
+  gameMode = GAME_MODE_BETTING_MODE;
+  currPlayer = 0;
+  currentPlayer = 0;
+  deck = buildDeck();
+  shuffleDeck(deck);
+  for (var i = 0; i < playerProfiles.length; i += 1) {
+    playerProfiles[i].playerHand = [];
+    playerProfiles[i].playerAceCount = 0;
+    playerProfiles[i].playerHandValue = 0;
+    playerProfiles[i].playerCurrRoundBet = 0;
+    playerProfiles[i].playerRoundScore = "";
+  }
+  dealerProfile = {
+    dealerHand: [],
+    dealerAceCount: 0,
+    dealerHandValue: 0,
+    dealerFinalValue: 0,
+  };
+}
+function totalReset() {
+  gameMode = GAME_MODE_CHOOSE_NUM_PLAYERS;
+  numPlayers = 0;
+  currPlayer = 0;
+  currentPlayer = 0;
+  input;
+  playerProfiles = [];
+  deck = [];
+  dealerProfile = {
+    dealerHand: [],
+    dealerAceCount: 0,
+    dealerHandValue: 0,
+    dealerFinalValue: 0,
+  };
+}
+
+///===================================================///
+///================= MAIN FUNCTION ===================///
+///===================================================///
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  console.log(deck.length);
+
+  if (gameMode == GAME_MODE_CHOOSE_NUM_PLAYERS) {
+    deck = buildDeck();
+    shuffleDeck(deck);
+    getNumPlayers(input);
+    return numPlayersMessage;
+  } else if (gameMode == GAME_MODE_BETTING_MODE) {
+    if (playerProfiles.length == 0) {
+      return "All players lost. No more players. Refresh the page to play again. ";
+    } else {
+      var bettingModeMessage = determineCurrentRoundBetSize(input);
```

Comment:
For the following lines, you are creating unnecessary variables. Just return the function directly. i.e. return determineCurrentRoundBetSize(input)

You should only create variables if you need the value else where